src/db/sysdb_ops.c

```diff
@@ -1914,15 +1914,17 @@ int sysdb_add_user(struct sss_domain_info *domain,
             goto done;
         }
 
-        ret = sysdb_search_group_by_gid(tmp_ctx, domain, uid, NULL, &msg);
-        if (ret != ENOENT) {
-            if (ret == EOK) {
-                DEBUG(SSSDBG_OP_FAILURE,
-                    "Group with GID [%"SPRIgid"] already exists in an "
-                    "MPG domain\n", gid);
-                ret = EEXIST;
+        if (uid != 0) { /* uid == 0 means non-POSIX object */
+            ret = sysdb_search_group_by_gid(tmp_ctx, domain, uid, NULL, &msg);
```

Comment:
I see it was like this before, but I wonder if  `uid` is the right parameter for a function searching by gid.

Reply:
Hi,

although it would be possible to use `gid` I think using `uid` better illustrates that this check is about users with an automatically created user-private-group. If you check the beginning of `sysdb_add_user()` use see that it is expected that for users the `gid` argument is `0` (it is set to the `uid` value then to make the following not full of `if` or `?:` but still `gid` is not something which stands for its own in this case).

bye,
Sumit